internal/apis/manifests/repositoryentry_types.go

```diff
@@ -0,0 +1,25 @@
+package manifests
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +kubebuilder:object:root=true
+type RepositoryEntry struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+
+	Data RepositoryEntryData
+}
+
+type RepositoryEntryData struct {
```

Comment:
Need a field to specify dependencies on other packages (including the required version range)